flink-end-to-end-tests/flink-end-to-end-tests-common/src/main/java/org/apache/flink/tests/util/cache/AbstractDownloadCache.java

```diff
@@ -91,6 +91,30 @@ public void afterTestSuccess() {
 
 	abstract boolean matchesCachedFile(Matcher matcher, String url);
 
+	private long getDownloadTimeoutMillis() {
```

Comment:
Let's refactor this to
a) be a `ParameterProperty<Duration>`
b) use `TimeUtils#parseDuration` for parsing
c) rename the property to `downloadTimeout` since with b) we can support any time unit

Reply:
Yeah,  good points.